app/models/session.rb

```diff
@@ -5,6 +5,13 @@ class Session < ApplicationRecord
   scope :alive, -> { where alive? }
   scope :expired, -> { where expired? }
 
+  def active?(providers = {})
+    return true unless self.expires_at
+    return false unless self.expires_at > Time.now.utc
+    return true if providers.empty?
+    providers.any? { |provider| self.data['provider'] == provider }
```

Comment:
The method `active?` accepts an optional argument (array) to check for specific providers. `providers.any?` is checking if the provider used to create the session is a valid provider, and if it is a valid provider, it returns true.